spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Graph.scala

```diff
@@ -62,6 +62,14 @@ class Graph[T: ClassTag](val inputs : Seq[ModuleNode[T]],
     extends Container[Activity, Activity, T]{
 
   override def updateOutput(input: Activity): Activity = {
+    val output = doForward(input) { (element, forwardInput) =>
+      element.updateOutput(forwardInput)
+    }
+    output
+  }
+
+  private def doForward(input: Activity)
```

Comment:
private def forwardExecution(action: (AbstractModule[Activity, Tensor[T], T], Activity) => Unit)(input: Activity): Activity

so you can def

private def doForward = forwardExecution((layer, input) => {layer.forward(input))
private def doUpdateOutput = forwardExecution((layer, input) => {layer.updateOutput(input)})

...
doFoward(node.element, ...)
doUpdateOutput(node.element, ...)